WASM browser test never runs: hardcoded v1.4.3 artifact name no longer exists

`WasmTest.yml`'s **Test WASM in Browser** job fails on every run that reaches it:

```
##[error]Unable to download artifact(s): Artifact not found for name: markdown-v1.4.3-extension-wasm_eh
```

The artifact name is hardcoded in two places (`.github/workflows/WasmTest.yml:33` and `:44`), both with the comment *"Use v1.4.3 extension to match duckdb-wasm 1.32.0 (stable release)"*. That pin was deliberate when written, but the distribution pipeline now builds against **v1.5.4**, so no `markdown-v1.4.3-*` artifact is produced and the download fails before any browser test runs. The summary step still prints "Extension built for: DuckDB v1.4.3" (`:261`).

## Why it matters

This job exists to guard the `LINKED_LIBS` class of WASM bug from #19 — a bug that shipped once already. It has not actually executed since the DuckDB target moved.

The main distribution pipeline **does** still build all three WASM variants and run the symbol-resolution check, so WASM is not unguarded — but the *browser integration* half, which is what caught #19's runtime failure mode, is dark.

## The real question

The pin exists because the extension and the `duckdb-wasm` npm package must agree on a DuckDB version. So this is not simply "change 1.4.3 to 1.5.4" — it needs the matching `duckdb-wasm` release for v1.5.4, and the two versions have to move together from now on.

Options, roughly in order of preference:

1. **Derive the version instead of hardcoding it.** Take it from the same place the pipeline does, so the artifact name cannot drift from what was built. Pair it with whichever `duckdb-wasm` version corresponds, and fail loudly if no matching package exists rather than silently skipping.
2. Bump both pins together to v1.5.4 and the matching `duckdb-wasm`, accepting that this recurs on the next DuckDB bump.
3. If no `duckdb-wasm` release matches v1.5.4 yet, make the job **skip explicitly with a stated reason** rather than fail on a missing artifact — a job that cannot run should say so, not look broken.

## Related

This is the third dark-test finding in this repo in a short span — after #36 (`require json` skipping five assertions, four unrelated to json) and the DuckDB-`main` canary being indistinguishable from the stable build in the checks list. Each was individually reasonable and collectively taught "red is normal here". A sweep for *jobs and files that appear to run but do not* seems worth doing once, deliberately.

--- a/.github/workflows/WasmTest.yml
+++ b/.github/workflows/WasmTest.yml
@@ -7,9 +7,14 @@ on:
   workflow_dispatch:
     inputs:
       duckdb_wasm_version:
-        description: 'duckdb-wasm npm version to test'
+        description: 'duckdb-wasm npm version to test (overrides the compatibility check)'
         required: false
-        default: '1.32.0'
+
+# The DuckDB version the distribution pipeline builds this extension against.
+# Keep in step with duckdb-stable-build in MainDistributionPipeline.yml -- the
+# artifact name is derived from it, and it decides which duckdb-wasm can load it.
+env:
+  EXTENSION_DUCKDB_VERSION: v1.5.4
 
 jobs:
   test-wasm-browser:
@@ -18,10 +23,38 @@ jobs:
     if: ${{ github.event_name == 'workflow_dispatch' || github.event.workflow_run.conclusion == 'success' }}
 
     steps:
+      # DuckDB refuses to load an extension built for a different version, so the
+      # duckdb-wasm bundle must embed exactly EXTENSION_DUCKDB_VERSION. The npm
+      # version does not encode that, so the pairing is recorded here explicitly.
+      #
+      #   duckdb-wasm 1.32.0 / 1.33.0  -> duckdb d1dc88f950 (v1.4.x)
+      #   duckdb-wasm 1.33.1-dev*      -> duckdb d8cdaa33fd (v1.5.5), built from main
+      #
+      # There is currently NO published duckdb-wasm carrying v1.5.4, so this job
+      # skips rather than failing. Add the pairing below when one ships, or pass
+      # duckdb_wasm_version on a manual run to override.
+      - name: Resolve a compatible duckdb-wasm
+        id: pair
+        run: |
+          override="${{ inputs.duckdb_wasm_version }}"
+          if [ -n "$override" ]; then
+            echo "wasm=$override" >> "$GITHUB_OUTPUT"
+            echo "compatible=true" >> "$GITHUB_OUTPUT"
+            echo "Using manually specified duckdb-wasm $override"
+            exit 0
+          fi
+          case "$EXTENSION_DUCKDB_VERSION" in
+            v1.4.3) echo "wasm=1.32.0" >> "$GITHUB_OUTPUT"; echo "compatible=true" >> "$GITHUB_OUTPUT" ;;
+            *)      echo "compatible=false" >> "$GITHUB_OUTPUT"
+                    echo "No published duckdb-wasm bundles DuckDB $EXTENSION_DUCKDB_VERSION; skipping." ;;
+          esac
+
       - name: Checkout repository
+        if: steps.pair.outputs.compatible == 'true'
         uses: actions/checkout@v4
 
       - name: Setup Node.js
+        if: steps.pair.outputs.compatible == 'true'
         uses: actions/setup-node@v4
         with:
           node-version: '20'
@@ -29,33 +62,34 @@ jobs:
       - name: Download WASM extension artifact
         uses: actions/download-artifact@v4
         with:
-          # Use v1.4.3 extension to match duckdb-wasm 1.32.0 (stable release)
-          name: markdown-v1.4.3-extension-wasm_eh
+          name: markdown-${{ env.EXTENSION_DUCKDB_VERSION }}-extension-wasm_eh
           path: wasm-ext
           github-token: ${{ secrets.GITHUB_TOKEN }}
           run-id: ${{ github.event.workflow_run.id }}
-        if: github.event_name == 'workflow_run'
+        if: steps.pair.outputs.compatible == 'true' && github.event_name == 'workflow_run'
 
       - name: Download WASM extension (manual trigger)
-        if: github.event_name == 'workflow_dispatch'
+        if: steps.pair.outputs.compatible == 'true' && github.event_name == 'workflow_dispatch'
         run: |
           mkdir -p wasm-ext
-          # Use v1.4.3 extension to match duckdb-wasm 1.32.0 (stable release)
-          gh run download --name markdown-v1.4.3-extension-wasm_eh --dir wasm-ext || echo "No artifact found, will skip extension test"
+          gh run download --name "markdown-${EXTENSION_DUCKDB_VERSION}-extension-wasm_eh" --dir wasm-ext || echo "No artifact found, will skip extension test"
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Create test directory
+        if: steps.pair.outputs.compatible == 'true'
         run: mkdir -p wasm-test
 
       - name: Install dependencies
+        if: steps.pair.outputs.compatible == 'true'
         working-directory: wasm-test
         run: |
           npm init -y
-          npm install @duckdb/duckdb-wasm@${{ inputs.duckdb_wasm_version || '1.32.0' }} playwright
+          npm install @duckdb/duckdb-wasm@${{ steps.pair.outputs.wasm }} playwright
           npx playwright install chromium
 
       - name: Copy extension to duckdb-wasm dist
+        if: steps.pair.outputs.compatible == 'true'
         run: |
           if [ -f wasm-ext/markdown.duckdb_extension.wasm ]; then
             cp wasm-ext/markdown.duckdb_extension.wasm wasm-test/node_modules/@duckdb/duckdb-wasm/dist/
@@ -65,6 +99,7 @@ jobs:
           fi
 
       - name: Create browser test
+        if: steps.pair.outputs.compatible == 'true'
         run: |
           cat > wasm-test/test-browser.mjs << 'TESTSCRIPT'
           import { chromium } from 'playwright';
@@ -249,6 +284,7 @@ jobs:
           TESTSCRIPT
 
       - name: Run browser tests
+        if: steps.pair.outputs.compatible == 'true'
         working-directory: wasm-test
         run: node test-browser.mjs
         timeout-minutes: 5
@@ -256,9 +292,20 @@ jobs:
       - name: Summary
         if: always()
         run: |
+          if [ "${{ steps.pair.outputs.compatible }}" != "true" ]; then
+            echo "## WASM Integration Test — SKIPPED" >> $GITHUB_STEP_SUMMARY
+            echo "" >> $GITHUB_STEP_SUMMARY
+            echo "No published \`duckdb-wasm\` bundles DuckDB ${{ env.EXTENSION_DUCKDB_VERSION }}, and DuckDB" >> $GITHUB_STEP_SUMMARY
+            echo "refuses to load an extension built for a different version, so there is no" >> $GITHUB_STEP_SUMMARY
+            echo "compatible pair to test. This is a skip, **not** a pass — the browser guard" >> $GITHUB_STEP_SUMMARY
+            echo "for the LINKED_LIBS bug (#19) is not running." >> $GITHUB_STEP_SUMMARY
+            echo "" >> $GITHUB_STEP_SUMMARY
+            echo "Add the pairing in the *Resolve a compatible duckdb-wasm* step when one ships." >> $GITHUB_STEP_SUMMARY
+            exit 0
+          fi
           echo "## WASM Integration Test Results" >> $GITHUB_STEP_SUMMARY
-          echo "- duckdb-wasm: ${{ inputs.duckdb_wasm_version || '1.32.0' }}" >> $GITHUB_STEP_SUMMARY
-          echo "- Extension built for: DuckDB v1.4.3" >> $GITHUB_STEP_SUMMARY
+          echo "- duckdb-wasm: ${{ steps.pair.outputs.wasm || 'n/a (skipped)' }}" >> $GITHUB_STEP_SUMMARY
+          echo "- Extension built for: DuckDB ${{ env.EXTENSION_DUCKDB_VERSION }}" >> $GITHUB_STEP_SUMMARY
           echo "- Test type: Browser (Playwright + Chromium)" >> $GITHUB_STEP_SUMMARY
           echo "" >> $GITHUB_STEP_SUMMARY
           echo "### Functions Tested" >> $GITHUB_STEP_SUMMARY

